worksheet.rb

```diff
@@ -31,4 +77,100 @@
 # - the total amount of money each driver has made
 # - the average rating for each driver
 # - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+# - Which driver has the highest average rating?
+
+# Define data structures
+average_ratings_array = []
+top_earning_drivers = []
+highest_rated_drivers = []
+
+# Define methods
+# Method to store the highest-achieving drivers in given categories
+def find_the_highest(array_to_compare, array_of_highest, hash)
+  
+  array_to_compare.each_with_index do |item, index|
+    if item == array_to_compare.max
+      array_of_highest << hash.keys[index]
+    end
+  end
+  
+end
+
+# Method to print the highest-achieving drivers in given categories
+def print_highest_results (array_of_highest, category_string, plural_string, singular_string)
```

Comment:
This is an extremely minor nitpick but: Ruby style guides would prefer if you didn't put a space between the method name and the opening parens, so: `def print_highest_results( array_of_highest ... )`